context.go

```diff
@@ -76,9 +76,7 @@ func getTChannelParams(ctx context.Context) *tchannelCtxParams {
 
 // NewContext returns a new root context used to make TChannel requests.
 func NewContext(timeout time.Duration) (context.Context, context.CancelFunc) {
-	return NewContextBuilder(timeout).
-		setSpan(NewRootSpan()).
```

Comment:
Sorry I should have been clearer. Returning `nil` isn't the issue, it's more that there's existing internal callers to `CurrentCall` that might be depending on the trace ID being set as they're about to make an outbound call.

I'll sync up over email about what existing callers are and we can figure out whether they need it to work or not.